yaksh/views.py

```diff
@@ -1,2955 +1,3004 @@
-            attempt_number)
-    writer = csv.writer(response)
-    if 'questions' in csv_fields:
-        csv_fields = _expand_questions(questions, csv_fields)
-    writer.writerow(csv_fields)
-
-    csv_fields_values = {
-            'name': 'user.get_full_name().title()',
-            'roll_number': 'user.profile.roll_number',
-            'institute': 'user.profile.institute',
-            'department': 'user.profile.department',
-            'username': 'user.username',
-            'marks_obtained': 'answerpaper.marks_obtained',
-            'out_of': 'question_paper.total_marks',
-            'percentage': 'answerpaper.percent',
-            'status': 'answerpaper.status'}
-    questions_scores = {}
-    for question in questions:
-        questions_scores['{0}-{1}'.format(question.summary, question.points)] \
-                = 'answerpaper.get_per_question_score({0})'.format(question.id)
-    csv_fields_values.update(questions_scores)
-
-    users = users.exclude(id=course.creator.id).exclude(
-        id__in=course.teachers.all())
-    for user in users:
-        row = []
-        answerpaper = None
-        papers = answerpapers.filter(user=user)
-        if papers:
-            answerpaper = papers.first()
-        for field in csv_fields:
-            try:
-                row.append(eval(csv_fields_values[field]))
-            except AttributeError:
-                row.append('-')
-        writer.writerow(row)
-    return response
-
-
-@login_required
-@email_verified
-def grade_user(request, quiz_id=None, user_id=None, attempt_number=None,
-               course_id=None):
-    """Present an interface with which we can easily grade a user's papers
-    and update all their marks and also give comments for each paper.
-    """
-    current_user = request.user
-    if not current_user.is_authenticated() or not is_moderator(current_user):
-        raise Http404('You are not allowed to view this page!')
-    course_details = Course.objects.filter(Q(creator=current_user) |
-                                           Q(teachers=current_user),
-                                           is_trial=False).distinct()
-    context = {"course_details": course_details}
-    if quiz_id is not None:
-        questionpaper_id = QuestionPaper.objects.filter(
-            quiz_id=quiz_id
-        ).values("id")
-        user_details = AnswerPaper.objects.get_users_for_questionpaper(
-            questionpaper_id, course_id
-        )
-        quiz = get_object_or_404(Quiz, id=quiz_id)
-        course = get_object_or_404(Course, id=course_id)
-        if not course.is_creator(current_user) and not \
-                course.is_teacher(current_user):
-            raise Http404('This course does not belong to you')
-
-        has_quiz_assignments = AssignmentUpload.objects.filter(
-                                question_paper_id=questionpaper_id
-                                ).exists()
-        context = {
-            "users"
```

Comment:
Please check why this humongous blob of change is showing up. Please undo and remove if this seems to be an error.